### PR TITLE
Workflow Notification Template Variables | LRDOCS-7311

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-7311, this includes variables from TemplateNotificationMessageGenerator, but I suspect there are more variables you can use (like ), and I think they come from NotificationMessageHelper. Let me know what you think, and then I'll bring it back into my queue and keep working on it.

### CODE REVIEW

File: developer/reference/articles/02-workflow/04-notifications.markdown [hunk 1]
```diff
@@ -147,6 +147,71 @@ attribute of the `<recipients>` tag as _To_, _CC_, or _BCC_.
...
+`KaleoInstanceToken`) are created each time a User clicks _Submit for
+Publication_. All of the attributes of the `KaleoInstanceTokenModel` are
+accessible in the notification template. For example,
+`${kaleoInstanceToken.modifiedDate?datetime}`.
+
+`${userId}` and `${userName}`
+: The User information returned is context dependent. If the notification is in
+a workflow task node, the User is retrieved from the `ServiceContext`.
+Otherwise, the User information is retrieved from the `KaleoInstanceToken`.
+Practically, this means that <!--What does it mean to an end user?-->.
```
Comment (rbohl) [inline developer/reference/articles/02-workflow/04-notifications.markdown:170]: Can give me some information about how this works?

File: developer/reference/articles/02-workflow/04-notifications.markdown [hunk 2]
```diff
@@ -147,6 +147,71 @@ attribute of the `<recipients>` tag as _To_, _CC_, or _BCC_.
...
+
+### Variables Available in Workflow Task Notifications
+
+These variables are available in notification templates that are inside a
+workflow task node:
+
+[`${kaleoTaskInstanceToken}`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/portal-workflow/portal-workflow-kaleo-api/src/main/java/com/liferay/portal/workflow/kaleo/model/KaleoTaskInstanceToken.java)
+: The task's token itself is available, with all of the
+`KaleoTaskInstanceTokenModel`'s attributes. For example,
```
Comment (rbohl) [inline developer/reference/articles/02-workflow/04-notifications.markdown:179]: Is this true? I'd like to tell workflow developers how to know which attributes or methods are available to call in their templates.

**Comment by inacionery** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: developer/reference/articles/02-workflow/04-notifications.markdown (hunk 2 above)
Comment (inacionery) [reply developer/reference/articles/02-workflow/04-notifications.markdown:179]: All methods are here https://docs.liferay.com/ce/apps/portal-workflow/latest/javadocs/com/liferay/portal/workflow/kaleo/model/KaleoTaskInstanceToken.html

**Comment by inacionery** [comment]
Pull request submitted to https://github.com/rbohl/liferay-docs/pull/152. [See changes here.](https://github.com/inacionery/liferay-docs/compare/bc9619a8c~3...bc9619a8c) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**